src/lib/rmt/maria_db/current_root_password_dialog.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2018 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require 'rmt/base'
+
+module RMT; end
+module RMT::MariaDB; end
+
+class RMT::MariaDB::CurrentRootPasswordDialog < RMT::Base
+  def run
+    ret = nil
+
+    UI.OpenDialog(
+      VBox(
+        VSpacing(1),
+        Heading(_('Database root password is required')),
+        VSpacing(1),
+        HBox(
+          HSpacing(2),
+          VBox(
+            Label(_('Please provide the current database root password.')),
+            MinWidth(15, Password(Id(:root_password), _('MariaDB root &password')))
+          ),
+          HSpacing(2)
+        ),
+        VSpacing(1),
+        HBox(
+          PushButton(Id(:cancel), Opt(:key_F9), Label.CancelButton),
+          HSpacing(2),
+          PushButton(Id(:ok), Opt(:default, :key_F10), Label.OKButton)
+        ),
+        VSpacing(1)
+      )
+    )
+
+    UI.SetFocus(Id(:root_password))
+
+    loop do
+      user_ret = UI.UserInput
+
+      if user_ret == :cancel
+        ret = nil
+        break
+      elsif user_ret == :ok
+        root_password = UI.QueryWidget(Id(:root_password), :Value)
+
+        if !root_password || root_password.empty?
+          UI.SetFocus(Id(:root_password))
+          Report.Error(_('Please provide the root password.'))
+          next
+        elsif !root_password_valid?(root_password)
+          UI.SetFocus(Id(:root_password))
+          Report.Error(_('The provided password is not valid.'))
+          next
+        end
+
+        ret = root_password
+        break
+      end
+    end
+
+    UI.CloseDialog
+
+    ret
+  end
+
+  def root_password_valid?(password)
+    run_command(
+      "echo 'show databases;' | mysql -u root -p%1 2>/dev/null",
+      password
```

Comment:
this will log password in log, which customer really does not like.

Reply:
How would you suggest to do it properly, then? Write to file first and then to pipe the file contents?

Reply:
recommended way is to write it to file. Possibility is also cheetah, where you can pass it to stdin like:

```
require "yast"

require "yast2/execute"

password = "secret"

out = Yast::Execute.on_target("cat", stdin: StringIO.new(password, "r"), stdout: :capture)

puts out.inspect
```

Reply:
second way I find safer as there is no risk of file collisions.

Reply:
@jreidinger MySQL doesn't allow for a password to be entered non-interactively, other than `mysql -u root --password=$(cat)` -- which gets escaped to `--password\=\$\(cat\)`.
I guess then file it is.

Reply:
@jreidinger does it really log the commands though? I've checked my `~/.y2log` and it seems the commands aren't logged there with the default log level.

Reply:
I am not sure, maybe with Y2DEBUG=1 it is logged, which sometimes we also ask.

Reply:
@jreidinger But with `Y2DEBUG=1` it logs even `SCR.Write` calls:

```
2018-03-02 12:32:00 <0> g87(32683) [scr] ScriptingAgent.cc(executeSubagentCommand):589 ScriptingAgent::executeSubagentCommand: Write
2018-03-02 12:32:00 <0> g87(32683) [scr] ScriptingAgent.cc(executeSubagentCommand):590 path: .target.string
2018-03-02 12:32:00 <0> g87(32683) [scr] ScriptingAgent.cc(executeSubagentCommand):591 arg: "/tmp/yast-test"
2018-03-02 12:32:00 <0> g87(32683) [scr] ScriptingAgent.cc(executeSubagentCommand):592 opt: "mysql -u root -pverysecretpassword"
2018-03-02 12:32:00 <0> g87(32683) [agent-system] ../../libscr/src/include/scr/Y2AgentComponent.h(evaluate):95 evaluate (`Write (.string, "/tmp/yast-test", "mysql -u root -pverysecretpassword"))
```

It doesn't look any different from `SCR.Execute`:

```
2018-03-02 12:33:42 <0> g87(334) [scr] ScriptingAgent.cc(executeSubagentCommand):589 ScriptingAgent::executeSubagentCommand: Execute
2018-03-02 12:33:42 <0> g87(334) [scr] ScriptingAgent.cc(executeSubagentCommand):590 path: .target.bash
2018-03-02 12:33:42 <0> g87(334) [scr] ScriptingAgent.cc(executeSubagentCommand):591 arg: "mysql -u root -pverysecretpassword"
2018-03-02 12:33:42 <0> g87(334) [scr] ScriptingAgent.cc(executeSubagentCommand):592 opt: null
2018-03-02 12:33:42 <0> g87(334) [agent-system] ../../libscr/src/include/scr/Y2AgentComponent.h(evaluate):95 evaluate (`Execute (.bash, "mysql -u root -pverysecretpassword"))
```

With the default log level it doesn't log neither `Write` nor `Execute`.
I could of course use Cheetah + `StringIO` as stdin, but is that really that big of a concern?